### ISSUE TITLE
Display title and description from EXIF tags

### ISSUE THREAD

**Issue by cpence** [thumbsup/thumbsup#102]
Since lightgallery supports it, it would be great if thumbsup would extract title and description for each image from the EXIF tags. fgallery, for example, uses the EXIF 'Title' and 'Description' tags.

**Comment by rprieto** [comment]
Hi @cpence, thanks for the issue. Thumbsup should already do this out of the box. Do you have an example of a file that's not read properly? The associated code is

```js
  return picasaValue(picasa, 'caption') ||
         tagValue(exif, 'EXIF', 'ImageDescription') ||
         tagValue(exif, 'IPTC', 'Caption-Abstract') ||
         tagValue(exif, 'IPTC', 'Headline') ||
         tagValue(exif, 'XMP', 'Description') ||
         tagValue(exif, 'XMP', 'Title') ||
         tagValue(exif, 'XMP', 'Label')
```

**Comment by rprieto** [comment]
Edit: sorry maybe I misunderstood. Thumbsup only reads the data as 1 field ("caption"). Do you suggest we read a title + description separately?

**Comment by rprieto** [comment]
In the current themes they are displayed in the light-box viewer, under the current photo. None of the demo images have captions unfortunately.

There's work underway to create more themes. A new theme for example could show the caption under each photo in the grid, like this example from https://codecanyon.net/item/jquery-thumbnail-gallery-with-lightbox/2711906:

![image](https://user-images.githubusercontent.com/189105/40747975-4e515706-645f-11e8-8b21-137ff9974a44.png)

**Comment by cpence** [comment]
Okay – I wasn't seeing them, but let me go away and tinker for a bit and see if I can make them show up as the code currently sits. I'll close this for now since it may well be my fault, will reopen if I can't get it working!

**Comment by rprieto** [comment]
Thanks, definitely let me know! You can keep an eye on https://github.com/thumbsup/thumbsup/issues/97 too if you're interested in new themes.

**Comment by mariaro** [comment]
Hello, 
I can also reproduce the issue that @cpence detailed: the caption is correctly read from, for example, IPTC Headline field, but it is not generated in the resulting index.html file. There is a small bug in the thumbnail.hbs template file.

For a npm package installation (version 2.3.1 as I installed today), a quick fix is to change the C:\Users\<username_here>\AppData\Roaming\npm\node_modules\thumbsup\templates\thumbnail.hbs file at approximately line 25. Just change the line 
` data-sub-html="{{caption}}"`
to
` data-sub-html="{{meta.caption}}"`
I added "meta." before "caption".

The corresponding file in the github repository: thumbsup\themes\base\partials\thumbnail.hbs (line 25 according to commit 03a9c9a)

Not the least, thank you very much @rprieto for a great gallery generator! It does exactly what I was looking for.
